src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateHoldingEventHandler.java

```diff
@@ -201,9 +207,28 @@ private void prepareDataAndReInvokeCurrentHandler(DataImportEventPayload dataImp
     });
   }
 
-  private void constructDataImportEventPayload(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, HoldingsRecord holding) {
-    dataImportEventPayload.getContext().put(HOLDINGS.value(), Json.encodePrettily(holding));
-    future.complete(dataImportEventPayload);
-    future.complete(dataImportEventPayload);
+  private void constructDataImportEventPayload(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, ItemCollection itemCollection, HoldingsRecord holding) {
+    HashMap<String, String> payloadContext = dataImportEventPayload.getContext();
+    payloadContext.put(HOLDINGS.value(), Json.encodePrettily(holding));
+    if(payloadContext.containsKey(ITEM.value())) {
```

Comment:
Can we move this line ItemCollection itemCollection = storage.getItemCollection(context) to this if statement?
We don't need to make additional call to storage if payload context does not contain ITEM

Reply:
ready